lib/src/builder/parsing/members_from_ast.dart

```diff
@@ -429,26 +426,54 @@ class _BoilerplateMemberDetector {
     return false;
   }
 
+  /// UiComponent, UiComponent2, UiStatefulComponent, FluxUiComponent, CustomUiComponent, ...
+  static final _componentBaseClassPattern = RegExp('Ui\w+Component');
+
   bool _detectPotentialComponent(ClassishDeclaration classish) {
     // Don't detect react-dart components as boilerplate components, since they cause issues with grouping
     // if they're in the same file as an OverReact component with non-matching names.
     if (!const {'Component', 'Component2'}.contains(classish.superclass?.nameWithoutPrefix)) {
       if (classish.name.name.endsWith('Component') ||
           classish.allSuperTypes
               .map((t) => t.typeNameWithoutPrefix)
-              .any(const {'UiComponent', 'UiComponent2'}.contains) ||
+              .any(_componentBaseClassPattern.hasMatch) ||
           (classish.superclass?.typeArguments?.arguments
                   ?.map((t) => t.typeNameWithoutPrefix)
                   ?.any(propsOrMixinNamePattern.hasMatch) ??
               false)) {
-        onComponent(BoilerplateComponent(classish, VersionConfidences.all(Confidence.neutral)));
+        const mixinBoilerplateBaseClasses = {
+          'UiComponent2',
+          'UiStatefulComponent2',
+          'FluxUiComponent2',
+          'FluxUiStatefulComponent2'
+        };
+        final confidences = VersionConfidences(
+          // If the component extends from a base class known to be supported by the new boilerplate,
+          // has no annotation, is not abstract, and does not have $isClassGenerated, then it's
+          // most likely intended to be part of a new boilerplate class component declaration.
+          //
+          // Make this `likely` so that components that don't get associated with factory/props
+          // due to naming issues aren't silently ignored.
+          v4_mixinBased: !classish.hasAbstractKeyword &&
+                  !_overridesIsClassGenerated(classish) &&
+                  mixinBoilerplateBaseClasses.contains(classish.superclass?.nameWithoutPrefix)
+              ? Confidence.likely
+              : Confidence.neutral,
+          v2_legacyBackwardsCompat: Confidence.neutral,
+          v3_legacyDart2Only: Confidence.neutral,
```

Comment:
This is really cool

Reply:
Thanks! I'm super happy that the confidence approach made this kind of change easy